yt/data_objects/construction_data_containers.py

```diff
@@ -816,6 +822,11 @@ def shape(self):
 
     def _setup_data_source(self):
         self._data_source = self.ds.region(self.center, self.left_edge, self.right_edge)
+        if self._base_data_source is not None:
+            # intersect the extra with the region
+            self._data_source = self.ds.intersection(
+                [self._base_data_source, self._data_source]
+            )
```

Comment:
So the current test failure is because YTArbitraryGrid does not call `super().__init__`. I'm hesitant to add `_base_data_source` to their common parent (YTSelectionContainer3D) because I'm worried it will get confusing...  

So I could either store the `data_source` kwarg  in `_data_source` and rewrite this bit to be:

```python
        reg = self.ds.region(self.center, self.left_edge, self.right_edge)
        if self._data_source is not None:
            # intersect the extra with the region
            self._data_source = self.ds.intersection([self._data_source, reg])
        else:
            self._data_source = reg
```

OR I could add a `_base_data_source` to YTArbitraryGrid (and set it to `None` and not add it as a kwarg there) 

OR I could have a type or attribute check here.

 I'm inclined to go with the first option -- @matthewturk any feeling here?

Reply:
First option is definitely my preference.  This change should currently be restricted.

Reply:
>  This change should currently be restricted. 

Ok, just pushed up the first option. Not sure what you meant here though -- was it just that it should only work for the covering (and not arbitrary) grid?

Reply:
I think it could work for AG but I want this PR not to expand too an uncontrollable size.  :)

Reply:
first option is in